offchain/danaswap-api/test/TestUtil.purs

```diff
@@ -14,29 +15,52 @@ import Contract.Monad (Contract, ContractEnv, withContractEnv)
 import Contract.Test.Plutip (PlutipConfig, runContractInEnv, withKeyWallet, withPlutipContractEnv)
 import Contract.Wallet (KeyWallet, privateKeysToKeyWallet)
 import Contract.Wallet.KeyFile (privatePaymentKeyFromFile, privateStakeKeyFromFile)
+import Ctl.Internal.Plutip.PortCheck (isPortAvailable)
+import Ctl.Util (withOurLogger)
 import Data.BigInt as BigInt
 import Data.Identity (Identity)
+import Data.List.Lazy (replicateM, toUnfoldable)
 import Data.UInt as UInt
+import Effect.Exception (throw)
+import Effect.Random (randomInt)
 import Node.Process (lookupEnv)
-import Test.Spec (Spec, SpecT, before)
-import Ctl.Util (withOurLogger)
+import Test.Spec (Spec, SpecT, before, parallel, sequential)
+import Test.Spec.Reporter (specReporter)
+import Test.Spec.Runner (defaultConfig, runSpec')
 
 data Mode = Local | Testnet
 type EnvRunner = (ContractEnv () -> KeyWallet -> Aff Unit) -> Aff Unit
 type EnvSpec = SpecT Aff EnvRunner Identity Unit
 
-getEnvRunner :: Mode -> Aff EnvRunner
-getEnvRunner Local = pure $ withPlutipContractEnv plutipConfig $ defaultWallet
+derive instance Eq Mode
+
+runOurSpec :: Mode -> Aff EnvRunner -> EnvSpec -> Aff Unit
```

Comment:
Add a docstring that this function decides depending on the mode whether the tests are run parallel or in sequence

Reply:
While writing the doc string here I realized the way I had it was kinda silly with getEnvRunner being exported and main has to use it but still pass the Mode here too so I pulled getEnvRunner into it and renamed it `runWithMode`.
I also cleaned up the module in general.